applications/OptimizationApplication/custom_utilities/response/mass_response_utils.h

```diff
@@ -0,0 +1,88 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Reza Najian Asl,
+//                   Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+// Application includes
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(OPTIMIZATION_APPLICATION) MassResponseUtils
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using GeometryType = ModelPart::ElementType::GeometryType;
+
+    ///@}
+    ///@name Static operations
+    ///@{
+
+    static void Check(const ModelPart& rModelPart);
+
+    static double CalculateValue(const ModelPart& rModelPart);
+
+    template<class TDataType>
+    static void CalculateSensitivity(
```

Comment:
The input should be a list of modelparts and a list of SensitivityVariable, due to the following facts:
1. since this function resets the rSensitivityVariable to zero at every call, and if the modelsparts have overlaps or common interfaces we will not get the full sensitivities on the interfaces. 
2. for adjoint-based sensitivities, we need a  list of rSensitivityVariable because there should be a single adjoint problem for multi controls like shape, thickness, .... If this function is called separately for every control variable then the adjoint problem should be solved at every call. This point is not applied to mass but in general the inputs should be lists for the sale of consistency and effieicny. 

The above points hold for the python implementation as well.